### PR TITLE
feat: adjust redis store exceptions

### PR DESCRIPTION
**Pull request by mattnichols**
Utilize Path Exceptions in redis store

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: store-redis/src/main/java/com/mx/redis/RedisStore.java [hunk 1]
```diff
@@ -42,83 +44,119 @@ public RedisStore(ObjectMap configurations) {
...
+  private <T> T safeCall(String operation, Function<StatefulRedisConnection<String, String>, T> runnable) {
+    try {
+      return runnable.apply(connection());
+    } catch (RedisStoreConnectionException e) {
+      throw e;
+    } catch (RedisException e) {
+      throw new RedisStoreOperationException("Redis error occurred on " + operation, e);
+    } catch (RuntimeException e) {
+      throw new RedisStoreOperationException("Unknown exception thrown by redis on " + operation, e);
+    }
```
Comment (stevecl5) [inline store-redis/src/main/java/com/mx/redis/RedisStore.java:159]: This is really cool!

File: store-redis/src/test/groovy/com/mx/redis/RedisStoreTest.groovy [hunk 2]
```diff
@@ -169,4 +172,49 @@ class RedisStoreTest extends Specification implements WithMockery {
     then:
     subject.status() == "OK"
   }
+
+  def "rethrows connection failure"() {
+    given:
+    subject.setConnection(null)
+    subject = spy(subject)
```
Comment (stevecl5) [inline store-redis/src/test/groovy/com/mx/redis/RedisStoreTest.groovy]: I usually add `spy` when instantiating the `subject`, so it doesn't need to be added to individual tests.

```
// in setup()
subject = spy(new RedisStore(configurations))
```

File: store-redis/src/main/java/com/mx/redis/RedisStore.java (hunk 1 above)
Comment (mattnichols) [reply store-redis/src/main/java/com/mx/redis/RedisStore.java:159]: 😎

File: store-redis/src/test/groovy/com/mx/redis/RedisStoreTest.groovy (hunk 2 above)
Comment (mattnichols) [reply store-redis/src/test/groovy/com/mx/redis/RedisStoreTest.groovy]: Fixed

**Review by stevecl5 — APPROVED** [review]

**Pull request merged**